openretina/modules/readout/multi_readout.py

```diff
@@ -28,27 +28,40 @@ def __init__(
         readout_reg_avg: bool = False,
     ):
         super().__init__()
+        self.session_init_args = {
+            "in_shape": in_shape,
+            "gaussian_mean_scale": gaussian_mean_scale,
+            "gaussian_var_scale": gaussian_var_scale,
+            "positive": positive,
+            "scale": scale,
+            "bias": bias,
+        }
+
+        self.add_sessions(n_neurons_dict)
+
+        self.gamma_readout = gamma_readout
+        self.gamma_masks = gamma_masks
+        self.gaussian_masks = gaussian_masks
+        self.readout_reg_avg = readout_reg_avg
+
+    def add_sessions(self, n_neurons_dict: dict[str, int]) -> None:
+        """Adds new sessions to the readout wrapper.
+        Can be called to add new sessions to an existing readout wrapper."""
+
+        assert all(key not in self.keys() for key in n_neurons_dict), (
+            "Found duplicate sessions in n_neurons_dict. Make sure to use different session names for each session."
+        )
```

Comment:
```suggestion
        if any(key in self.keys() for key in n_neurons_dict):
            duplicate_session_names = set(self.keys()).intersection(n_neuron_dict.keys())
            raise ValueError(f"Found duplicate sessions in n_neurons_dict:  {duplicate_session_names=}. Make sure to use different session names for each session."
        )
```

Normally you use assert to check the internal consistency within a class. As this is also called from other places, I'd prefer raising an error. The other suggestion is to also print duplicate sessions in the error.